### PR TITLE
Real mocks for OpenTable + Validation coverage

### PR DESCRIPTION
**Pull request by fergusdixon**
## Overview

| Q                 | A
| ----------------- | ---
| Bugfix?           | yes
| New feature?      | yes
| Breaking Changes? | no
| Fixed issues      | #11 

## The problem
We lacked real test mocks for pulling table data, and coverage was missing for the Validation class
...

## How I resolved it
Added mocks + tests
...

## Notes
Further testing for posting payments is required, however we need to wait for grant to get back to us about that before I can continue
...

Notify the following people: @darrynten

### CODE REVIEW

File: src/Validation.php [hunk 1]
```diff
@@ -41,22 +41,6 @@ public function isValidPrimitive($item, $definedType)
     }
 
     /**
-     * Validates an enumurable list
-     *
-     * @param string $value
-     * @param string $regex
-     */
-    public function validateEnum($value, $validValues)
```
Comment (fergusdixon) [inline src/Validation.php:49]: There are no enumerable lists in Pilot?

**Review by darrynten — APPROVED** [review]
This is great :)

Can we get this in and then can we get real mocks for some different types of vendor payments?

File: tests/PilotLive/ValidationTest.php [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            ValidationException::class,
+            sprintf("Validation error value %s String length is out of range", $message),
+            ValidationException::STRING_LENGTH_OUT_OF_RANGE
+        );
+        $this->validateRange($test, $min, $max);
+    }
+    public static function stringExceptionProvider()
+    {
+        return array(
+            array(
```
Comment (darrynten) [inline tests/PilotLive/ValidationTest.php]: You should use short array syntax always.

These can be replaced with `[]` notation.

**Review by darrynten — CHANGES_REQUESTED** [review]

File: tests/PilotLive/ValidationTest.php (hunk 2 above)
Comment (fergusdixon) [reply tests/PilotLive/ValidationTest.php]: Fixed 👍

**Review by darrynten — APPROVED** [review]

**Pull request merged**